interpreter/cling/lib/Interpreter/Interpreter.cpp

```diff
@@ -1026,52 +1028,25 @@ namespace cling {
   Interpreter::codeComplete(const std::string& line, size_t& cursor,
                             std::vector<std::string>& completions) const {
 
-    const char * const argV = "cling";
     std::string resourceDir = this->getCI()->getHeaderSearchOpts().ResourceDir;
     // Remove the extra 3 directory names "/lib/clang/3.9.0"
     StringRef parentResourceDir = llvm::sys::path::parent_path(
                                   llvm::sys::path::parent_path(
                                   llvm::sys::path::parent_path(resourceDir)));
     std::string llvmDir = parentResourceDir.str();
 
-    Interpreter childInterpreter(*this, 1, &argV, llvmDir.c_str());
-    if (!childInterpreter.isValid())
-      return kFailure;
-
-    auto childCI = childInterpreter.getCI();
-    clang::Sema &childSemaRef = childCI->getSema();
-
-    // Create the CodeCompleteConsumer with InterpreterCallbacks
-    // from the parent interpreter and set the consumer for the child
-    // interpreter.
-    ClingCodeCompleteConsumer* consumer = new ClingCodeCompleteConsumer(
-                getCI()->getFrontendOpts().CodeCompleteOpts, completions);
-    // Child interpreter CI will own consumer!
-    childCI->setCodeCompletionConsumer(consumer);
-    childSemaRef.CodeCompleter = consumer;
-
-    // Ignore diagnostics when we tab complete.
-    // This is because we get redefinition errors due to the import of the decls.
-    clang::IgnoringDiagConsumer* ignoringDiagConsumer =
-                                            new clang::IgnoringDiagConsumer();
-    childSemaRef.getDiagnostics().setClient(ignoringDiagConsumer, true);
-    DiagnosticsEngine& parentDiagnostics = this->getCI()->getSema().getDiagnostics();
-
-    std::unique_ptr<DiagnosticConsumer> ownerDiagConsumer =
-                                                parentDiagnostics.takeClient();
-    auto clientDiagConsumer = parentDiagnostics.getClient();
-    parentDiagnostics.setClient(ignoringDiagConsumer, /*owns*/ false);
-
-    // The child will desirialize decls from *this. We need a transaction RAII.
-    PushTransactionRAII RAII(this);
+    // arguments for constructing CI
+    auto declCollector = std::make_unique<cling::DeclCollector>();
+    const ModuleFileExtensions& moduleExtensions = {};
 
-    // Triger the code completion.
-    childInterpreter.CodeCompleteInternal(line, cursor);
+    auto InterpCI = std::unique_ptr<clang::CompilerInstance>(
+        CIFactory::createCI("\n", getOptions(), llvmDir.c_str(),
+                            std::move(declCollector), moduleExtensions,
```

Comment:
Not sure if it would be possible to make the `ASTConsumer` optional if we don't really need it... but ok for the current patch